### PR TITLE
update otel-core

### PR DESCRIPTION
**Pull request by codeboten**
This includes:
* rename from instrumentation library to instrumentation scope.
* removal of `componenthelper.New`

### CODE REVIEW

**Review by mx-psi — APPROVED** [review]

File: exporter/newrelicexporter/transformer.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func newTransformer(logger *zap.Logger, buildInfo *component.BuildInfo, details
 	return &transformer{logger: logger, OverrideAttributes: overrideAttributes, details: details}
 }
 
-func (t *transformer) CommonAttributes(resource pdata.Resource, lib pdata.InstrumentationLibrary) map[string]interface{} {
+func (t *transformer) CommonAttributes(resource pdata.Resource, lib pdata.InstrumentationScope) map[string]interface{} {
```
Comment (Aneurysm9) [inline exporter/newrelicexporter/transformer.go:66]: This changes the signature of an exported method (though on an unexported type).  Do we know whether this is supposed to be implementing an interface?

**Review by pmm-sumo — APPROVED** [review]
Phew, that's quite a changeset!

File: exporter/newrelicexporter/transformer.go (hunk 1 above)
Comment (codeboten) [reply exporter/newrelicexporter/transformer.go:66]: This is a good question, maybe @alanwest @jack-berg @nrcventura can help here.

Comment (alanwest) [reply exporter/newrelicexporter/transformer.go:66]: This change looks safe.

Comment (nrcventura) [reply exporter/newrelicexporter/transformer.go:66]: I agree that it looks safe. That exported method is not implementing an interface. The transformer type is only meant to be used internally.

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**